### PR TITLE
Add `Unbounded` to PosAngleConstraint

### PR DESCRIPTION
**Pull request by toddburnside**
Will be merged as part of the grand transition to using the new ODB.

### CODE REVIEW

**Review by swalker2m — APPROVED** [review]
Looks good to me. :+1:  Seems like @tpolecat mentioned that there's no longer a need for widening constructors but I'm not certain.

File: modules/core/shared/src/main/scala/lucuma/core/model/PosAngleConstraint.scala [hunk 1]
```diff
@@ -20,6 +20,22 @@ sealed trait PosAngleConstraint extends Product with Serializable
 
 object PosAngleConstraint extends PosAngleConstraintOptics {
 
+  /**
+   * Specifies that the there is no constraint on the position angle.
+   * It may be set to any angle for this observation.
+   */
+  case object Unbounded extends PosAngleConstraint {
```
Comment (cquiroz) [inline modules/core/shared/src/main/scala/lucuma/core/model/PosAngleConstraint.scala:27]: We used to have in most places `Option[PosAngleConstraint]` where `None` would be used as `UNBOUNDED`
It maybe worth check on lucuma-core where this pattern has been used and fix it

Comment (toddburnside) [reply modules/core/shared/src/main/scala/lucuma/core/model/PosAngleConstraint.scala:27]: I don't know of any places in lucuma-core, but I'll look. I have another pull request coming for `lucuma-catalog`. Plus, of course, the big PR for explore.

Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/model/PosAngleConstraint.scala:27]: Right, we may encode that on catalog or explore

Comment (toddburnside) [reply modules/core/shared/src/main/scala/lucuma/core/model/PosAngleConstraint.scala:27]: The encoder is in lucuma-schemas

**Pull request merged**